### ISSUE TITLE
Receiving one value out of two from /stage/state/needle_pose

### ISSUE THREAD

**Issue by lisaepfl** [SmartNeedle/ros2_needle_shape_publisher#11]
It seems like the NeedleNode is receiving only one value out of two of the ones that the NeedleGuide publishes under "/stage/state/needle_pose". For example it receives y=2, then y=4, y=6 etc but I am publishing y= 1, y=2, y=3, y=4...

**Comment by lisaepfl** [comment]
I checked again and needle_shape_publishing receives every value that NeedleGuide sends but with a delay. NeedleGuide is publishing at a 50ms frequency. I tried changing it but it doesn't seem to solve the issue.

**Comment by dlezcan1** [comment]
Yes, I think it’s a numerical issue where the publishing is limited by how fast the computations can take place. It can’t be any faster until the code is further optimized. At the moment, it’s limited to the current publishing rate and is variable based on the complexity of the shape-sensing
